### PR TITLE
Use Otelcol v0.3 and send spans directly

### PR DESCRIPTION
**Pull request by pmm-sumo**
###### Description

Use OpenTelemetry Collector @ Sumo v0.3 series, which includes sources processor (tagging `_sourceCategory` and `_sourceName`) and sends traces directly

###### Testing performed

- [ ] ci/build.sh
- [x] Redeploy fluentd and fluentd-events pods
- [x] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -630,15 +636,13 @@ otelcol:
...
-        # Number of tickers that loop over batch buckets
-        num_tickers: 10
         # Time duration after which a batch will be sent regardless of size
         timeout: 5s
     extensions:
       health_check: {}
     exporters:
       zipkin:
-        url: "exporters.zipkin.url_replace"
+        url: {{ .Values.sumologic.traces.endpoint || quote }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:645]: helm templating doesn't work in `values.yaml`
Thats why there is string which is replaced later in `traces.conf` file

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -597,10 +603,10 @@ otelcol:
             #serviceName: service
             statefulSetName: statefulset
           annotations:
-            - tag_name: pod_annotation_%s
+            - tag_name: "annotation:%s"
               key: "*"
           labels:
-            - tag_name: pod_label_%s
+            - tag_name: "label:%s"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:609]: we starting to be inconsistent with the rest of the solution

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -538,8 +538,8 @@ otelcol:
     memBallastSizeMib: "683"
     image:
       name: "sumologic/opentelemetry-collector"
-      tag: "0.2.7.1"
-      pullPolicy: IfNotPresent
+      tag: "0.3.0.3"
+      pullPolicy: Always
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:542]: `pullPolicy` should be still `IfNotPresent`

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:645]: also `||` -> `|`

File: deploy/helm/sumologic/values.yaml (hunk 2 above)
Comment (pmm-sumo) [reply deploy/helm/sumologic/values.yaml:611]: Or it's the opposite, e.g. https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/master/fluent-plugin-kubernetes-sumologic/lib/fluent/plugin/filter_kubernetes_sumologic.rb#L52

I am going to prepare a summary later and run it by the team. There are several alternative naming solutions being used across the code

File: deploy/helm/sumologic/conf/traces/traces.otelcol.conf.yaml [hunk 4]
```diff
@@ -1,4 +1,7 @@
 {{- $yamlFile := toYaml $.Values.otelcol.config }}
-{{- $fullname := include "sumologic.fullname" . }}
-{{- $url := list "http://" $fullname ":9411/api/v2/spans" | join "" }}
-{{- $yamlFile | replace "exporters.zipkin.url_replace" $url | nindent 2 }}
+{{- $tracesEndpoint := include "sumologic.traces.endpoint" . | quote }}
+{{- $sourceName := include "sumologic.sourceName" . | quote }}
+{{- $sourceCategory := include "sumologic.sourceCategory" . | quote }}
+{{- $sourceCategoryPrefix := include "sumologic.sourceCategoryPrefix" . | quote }}
+{{- $sourceCategoryReplaceDash := include "sumologic.sourceCategoryReplaceDash" . | quote }}
+{{- $yamlFile | replace "exporters.zipkin.url_replace" $tracesEndpoint | replace "processors.source.name.replace" $sourceName | replace "processors.source.category.replace" $sourceCategory | replace "processors.source.category_prefix.replace" $sourceCategoryPrefix | replace "processors.source.category_replace_dash.replace" $sourceCategoryReplaceDash | nindent 2 }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/traces/traces.otelcol.conf.yaml:7]: I believe there is no need to use `_helpers` in this case
`replace "processors.source.name.replace" $sourceName` could be just `replace "processors.source.name.replace" .Values.sumologic.sourceName`

File: deploy/helm/sumologic/values.yaml [hunk 5]
```diff
@@ -630,15 +638,13 @@ otelcol:
...
-        # Number of tickers that loop over batch buckets
-        num_tickers: 10
         # Time duration after which a batch will be sent regardless of size
         timeout: 5s
     extensions:
       health_check: {}
     exporters:
       zipkin:
-        url: "exporters.zipkin.url_replace"
+        url:  "exporters.zipkin.url_replace"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:647]: extra space before `"e`

File: deploy/helm/sumologic/values.yaml (hunk 2 above)
Comment (pmm-sumo) [reply deploy/helm/sumologic/values.yaml]: Changed it back to `pod_labels_%s`

File: deploy/helm/sumologic/conf/traces/traces.otelcol.conf.yaml [hunk 6]
```diff
@@ -1,4 +1,8 @@
...
-{{- $fullname := include "sumologic.fullname" . }}
-{{- $url := list "http://" $fullname ":9411/api/v2/spans" | join "" }}
-{{- $yamlFile | replace "exporters.zipkin.url_replace" $url | nindent 2 }}
+{{- $_collector := .Values.sumologic.collectorName | default .Values.sumologic.clusterName | quote }}
+{{- $endpoint := .Values.sumologic.traces.endpoint | quote }}
+{{- $sourceName := .Values.sumologic.sourceName | quote }}
+{{- $sourceCategory := .Values.sumologic.sourceCategory | quote }}
+{{- $sourceCategoryPrefix := .Values.sumologic.sourceCategoryPrefix | quote }}
+{{- $sourceCategoryReplaceDash := .Values.sumologic.sourceCategoryReplaceDash | quote }}
+{{- $yamlFile | replace "processors.source.collector.replace" $_collector | replace "exporters.zipkin.url_replace" $endpoint | replace "processors.source.name.replace" $sourceName | replace "processors.source.category.replace" $sourceCategory | replace "processors.source.category_prefix.replace" $sourceCategoryPrefix | replace "processors.source.category_replace_dash.replace" $sourceCategoryReplaceDash | nindent 2 }}
```
Comment (perk-sumo) [inline deploy/helm/sumologic/conf/traces/traces.otelcol.conf.yaml:8]: So it turns out it's possible to have variables in the values.yaml file 😉 👏

Comment (pmm-sumo) [reply deploy/helm/sumologic/conf/traces/traces.otelcol.conf.yaml:8]: Yes, per @sumo-drosiek suggestion :) Thank you!

**Pull request merged**